lib/sinatra/base.rb

```diff
@@ -181,7 +181,7 @@ def drop_body?
   class ExtendedRack < Struct.new(:app)
     def call(env)
       result, callback = app.call(env), env['async.callback']
-      return result unless callback and async?(*result)
+      return result unless callback && async?(*result)
```

Comment:
I believe the intended meaning here is `and` not `&&`, since this is composing actions, not evaluating a boolean expression (though, to be fair the `async?` at the end makes this a bit ambiguous).